graphql_compiler/compiler/compiler_frontend.py

```diff
@@ -709,6 +710,40 @@ def _preprocess_graphql_string(graphql_string):
     return graphql_string + '\n'
 
 
+def _validate_schema_and_ast(schema, ast):
+    """Validate the supplied graphql schema and ast.
+
+    This method wraps around graphql-core's validation to enforce a stricter requirement of the
+    schema-- all directives supported by the compiler must be declared by the schema, regardless of
+    whether each directive is used in the query or not.
+
+    Args:
+        schema: GraphQL schema object, created using the GraphQL library
+        ast: abstract syntax tree representation of a graphql query
+
+    Returns:
+        list containing schema and/or query validation errors
+    """
+    core_graphql_errors = validate(schema, ast)
+
+    # Extract name, locations, and args keys in order to compare schema directives and directives
+    # which are supported by the graphql compiler.
+    graphql_compiler_directives = [(s.name, s.locations, s.args.keys()) for s in DIRECTIVES]
```

Comment:
I'm a bit concerned about `s.args.keys()` since this code is meant to be py2 and py3 at the same time. Also, GraphQL directives don't prescribe argument order (unfortunately).

My recommendation would be to use `set(s.args.keys())`.

I would also make the list comprehension multi-line just like the one on 735-738, and maybe also use `directive` as the variable of the comprehension, for the sake of consistency.